### PR TITLE
feat(nova-4): Adds support for Nova v4.x

### LINKED ISSUE #47: Support for Nova v4.x
Since this repository hasn't been updated since March 2020, and current version does not support Laravel Nova v4.x, I've forked the repo and made necessary changes.

Primary change: fixed Comment Resource to match the new Nova 4 signature with NovaRequest instead of Request
Description of the major changes from Nova 4: https://nova.laravel.com/docs/4.0/upgrade.html#nova-request

My fork of nova-comments with support for Nova 4: https://github.com/amukhin-cyberhull/nova-comments

### PR DESCRIPTION
**Pull request by chrillep**
closes #47 

TODO:

- [x] Can you update CHANGELOG.md (I'm thinking we'll probably tag a v2 or what do you think about a v4.0 to match Nova major version?)
- [x] Can you update README.md, under "## Requirements": "This Nova resource tool requires Nova 4.0 or higher (if you are using an older version of Nova, use the tagged version [1.0.2](https://github.com/kirschbaum-development/nova-comments/releases/tag/1.0.2))."
- [x] Can you remove the date-format key/value from the nova-comments.php config file since it's no longer being used?

### CODE REVIEW

**Review by adammparker — CHANGES_REQUESTED** [review]
Hello @chrillep!

First - thank you so much for contributing to the package! Just this morning I started working on the upgrade and got through most of it and pushed up a branch before realizing that you had already opened this PR. 🔥 

I've left a couple notes for your review, I can also jump in and make a couple of the tweaks I'm recommending if you prefer/don't have time.

A couple of additional things that didn't fit in as comments directly in the PR code:
1) Can you update CHANGELOG.md (I'm thinking we'll probably tag a v2 or what do you think about a v4.0 to match Nova major version?)
2) Can you update README.md, under "## Requirements": _"This Nova resource tool requires Nova 4.0 or higher (if you are using an older version of Nova, use the tagged version [1.0.2](https://github.com/kirschbaum-development/nova-comments/releases/tag/1.0.2))."_
3) Can you remove the `date-format` key/value from the nova-comments.php config file since it's no longer being used?

**Comment by chrillep** [comment]
> Hello @chrillep!
> 
> 
> 
> First - thank you so much for contributing to the package! Just this morning I started working on the upgrade and got through most of it and pushed up a branch before realizing that you had already opened this PR. 🔥 
> 
> 
> 
> I've left a couple notes for your review, I can also jump in and make a couple of the tweaks I'm recommending if you prefer/don't have time.
> 
> 
> 
> A couple of additional things that didn't fit in as comments directly in the PR code:
> 
> 1) Can you update CHANGELOG.md (I'm thinking we'll probably tag a v2 or what do you think about a v4.0 to match Nova major version?)
> 
> 2) Can you update README.md, under "## Requirements": _"This Nova resource tool requires Nova 4.0 or higher (if you are using an older version of Nova, use the tagged version [1.0.2](https://github.com/kirschbaum-development/nova-comments/releases/tag/1.0.2))."_
> 
> 3) Can you remove the `date-format` key/value from the nova-comments.php config file since it's no longer being used?

first of thanks for taking the time.

second feel free to tweak/change anything as you see fit 😊.

i basicly cherry-picked @centgart fork.

I'll try and answer/fix the comments ASAP.

as for tag. yeah I would use v.4, as you said. it just makes more sense.

File: nova.mix.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class NovaExtension {
+  name() {
+    return 'nova-extension'
+  }
+
+  register(name) {
+    this.name = name
+  }
+
+  webpackConfig(webpackConfig) {
```
[the remark this answers is not in the document]
Comment (chrillep) [reply nova.mix.js:14]: yep implemented

File: resources/js/components/Comment.vue [hunk 2]
```diff
@@ -19,7 +19,8 @@
 </template>
 
 <script>
-    // require('moment-timezone');
+    require('lodash');
+    import moment from 'moment-timezone';
```
[the remark this answers is not in the document]
Comment (chrillep) [reply resources/js/components/Comment.vue]: deleted as imported via webpack

File: resources/js/components/Tool.vue [hunk 3]
```diff
@@ -26,7 +26,7 @@
                     On MacOS, press ⌘ + Enter to save
```
[the remark this answers is not in the document]
Comment (chrillep) [reply resources/js/components/Tool.vue:26]: fixed

File: resources/css/field.css [hunk 4]
```diff
@@ -0,0 +1 @@
+/* Nova Field CSS */
```
[the remark this answers is not in the document]
Comment (chrillep) [reply resources/css/field.css]: yep removed the field css

File: webpack.mix.js [hunk 5]
```diff
@@ -1,5 +1,11 @@
...
-    .js('resources/js/tool.js', 'js')
-    .sass('resources/sass/tool.scss', 'css')
+require('./nova.mix')
+
+mix
+  .setPublicPath('dist')
+  .js('resources/js/tool.js', 'js')
+  .vue({ version: 3 })
+  .css('resources/css/tool.css', 'css')
+  .css('resources/css/field.css', 'css')
```
[the remark this answers is not in the document]
Comment (chrillep) [reply webpack.mix.js]: i removed the field css 
the tool.css is from the tool stub, so not needed but keeps it more inline with what nova generates

File: src/Nova/Comment.php [hunk 6]
```diff
@@ -40,11 +41,11 @@ class Comment extends Resource
     /**
      * Get the fields displayed by the resource.
      *
-     * @param  \Illuminate\Http\Request  $request
+     * @param  \Nova\Http\Requests\NovaRequest  $request
```
[the remark this answers is not in the document]
Comment (chrillep) [reply src/Nova/Comment.php]: yep done

**Review by adammparker — APPROVED** [review]
Looks good, really appreciate the contribution @chrillep

**Pull request merged**